### PR TITLE
[terra-theme-provider] Add support for theme density

### PR DESCRIPTION
**Pull request by cm9361**
### Summary

**What was changed:**
Update the ThemeProvider to support specifying the density of the theme.

**Why it was changed:**
The change was made to styles that differed for a theme based on the data density.  This approach prevents the need to create a separate theme for each data density classification specified by our user experience team.

### Testing

This change was tested using:

- [ ] WDIO
- [X] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details
N/A

**This PR resolves:**

UXPLATFORM-9934 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-theme-provider/tests/jest/ThemeProvider.test.jsx [hunk 1]
```diff
@@ -14,9 +18,42 @@ describe('ThemeProvider', () => {
     expect(wrapper).toMatchSnapshot();
   });
 
-  it('should render a theme provider component', () => {
-    const wrapper = render(defaultRender);
-    expect(wrapper).toMatchSnapshot();
+  it('verify theme name is added to documentElement class list', () => {
```
Comment (sycombs) [inline packages/terra-theme-provider/tests/jest/ThemeProvider.test.jsx:21]: nit: can we word these so they sound "sentence-like", i.e. "it verifies theme name is added..."?

File: packages/terra-framework-docs/src/terra-dev-site/doc/theme-provider/example/DefaultThemeProvider.jsx [hunk 2]
```diff
@@ -38,6 +45,18 @@ class DefaultThemeProvider extends React.Component {
           </select>
         </div>
       );
+
+      densitySwitcher = (
+        <div className={cx('theme-switcher-wrapper')}>
+          <label htmlFor="density"> Theme: </label>
```
[the remark this answers is not in the document]
Comment (cm9361) [reply packages/terra-framework-docs/src/terra-dev-site/doc/theme-provider/example/DefaultThemeProvider.jsx:51]: Great catch.  Will update.

File: packages/terra-theme-provider/tests/jest/ThemeProvider.test.jsx (hunk 1 above)
Comment (cm9361) [reply packages/terra-theme-provider/tests/jest/ThemeProvider.test.jsx:21]: I will make this change.

**Review by sycombs — APPROVED** [review]

File: packages/terra-theme-provider/tests/jest/ThemeProvider.test.jsx (hunk 1 above)
Comment (nikhitasharma) [inline packages/terra-theme-provider/tests/jest/ThemeProvider.test.jsx:21]: Should this also be changed to "verifies ...." format like the other tests?

**Review by nikhitasharma — APPROVED** [review]

**Pull request merged**